### PR TITLE
Additional tenure update steps

### PR DESCRIPTION
**Pull request by adamtry**
This extends the tenure update script to also update the tenure in the property elasticsearch, as well as to go through each householdmember of a tenure and choose to update their tenure start/end dates.

### CODE REVIEW

File: aws/src/database/multi_table/amend_tenure.py [hunk 1]
```diff
@@ -11,17 +11,18 @@
 # Config - change these
 STAGE = Stage.HOUSING_PRODUCTION
 STAGE_PARAM = "production"  # Stage parameter for SSM
-ES_EXTENSION = "tenures"
-_update_date = datetime(day=31, month=1, year=2022).isoformat()
+_update_date = datetime(day=3, month=7, year=2023).isoformat()
```
Comment (Duslerke) [inline aws/src/database/multi_table/amend_tenure.py:14]: On line 19, if we take Tenure Id from user input, pehaps, we can also take the date time as well?
Seems a bit odd that one is hardcoded, while the other is not :thinking:

File: aws/src/database/multi_table/amend_tenure.py [hunk 2]
```diff
@@ -43,26 +44,24 @@
...
+    _confirm("Correct?")
+    tenure = get_tenure_dynamodb(TENURE_ID)
+    connect_to_jumpbox_for_es(instance_id=INSTANCE_ID, stage=STAGE.value)
+    # update_tenure_elasticsearch(tenure_pk=TENURE_ID)
+    update_property_elasticsearch(property_pk=tenure["tenuredAsset"]["id"])
     # update_tenure_dynamodb(tenure)
-    update_property_dynamodb(tenure)
-    # update_person_dynamodb(tenure)
+    # update_property_dynamodb(tenure)
+    # update_persons_dynamodb(tenure)
```
Comment (Duslerke) [inline aws/src/database/multi_table/amend_tenure.py:60]: When some value needs to update/removed within a record this has to be done across multiple storages. So from my understanding that's what all these comments are for.

Any chance they could get grouped by HALO ticket type? So, for example:
``` py
def remove_tenure_end_date(tenure):
    print(f"Removing end date from tenure (id): {tenure.id}")
    print("updating tenure table")
    update_tenure_dynamodb(tenure)
    print("updating property table")
    update_property_dynamodb(tenure)
    print("updating person table")
    update_persons_dynamodb(tenure)
    print("updating elastic search...")
    etc(...)
```

**Review by Duslerke — APPROVED** [review]

File: aws/src/database/multi_table/amend_tenure.py (hunk 1 above)
Comment (adamtry) [reply aws/src/database/multi_table/amend_tenure.py:14]: Yeah, you may be right. I was thinking about this but I'm not sure how to make the CLI input for the date object be user-friendly. Comma separated or something?

Comment (humulla) [reply aws/src/database/multi_table/amend_tenure.py:14]: I agree with this point as well

**Review by humulla — APPROVED** [review]

File: aws/src/database/multi_table/amend_tenure.py (hunk 2 above)
Comment (adamtry) [reply aws/src/database/multi_table/amend_tenure.py]: The script isn't as flexible as I'd like but it's difficult to make it flexible and keep it user-friendly and error-avoidant.

Comment (Duslerke) [reply aws/src/database/multi_table/amend_tenure.py]: > I'm not sure what's easiest for us to work with. I think in most cases we'd be running through every step but personally I get paranoid and like to run it bit by bit so I comment the other steps out.

Could have a "press enter to continue" or "continue? y/n" prompt before every step I suppose

File: aws/src/database/multi_table/amend_tenure.py (hunk 1 above)
Comment (Duslerke) [reply aws/src/database/multi_table/amend_tenure.py]: doesn't matter so long the prompt asking it tells what format to expect, ex:
> "Please enter new start date (format is mm/dd/yyyy): "

Comment (Duslerke) [reply aws/src/database/multi_table/amend_tenure.py]: sort of like AWS CLI configure command

**Review by humulla — COMMENTED** [review]
Another thing - we need to ensure that the order of the script is to first update Dynamodb and the elastic search, as ES syncs Dynamodb data it helps to keep things consistent if Dynamodb is updated first.

File: aws/src/database/multi_table/amend_tenure.py (hunk 1 above)
Comment (adamtry) [reply aws/src/database/multi_table/amend_tenure.py]: The current solution includes a text parser, so most formats should work fine. DD/MM/YYYY or DD-MM-YYYY or YYYY-MM-DD, etc. I tested it, it's very flexible. It's set to expect days before months in the day-month-year formats.

Comment (adamtry) [reply aws/src/database/multi_table/amend_tenure.py]: There's also a confirm step right after it parses and converts the date.

**Comment by adamtry** [comment]
> Another thing - we need to ensure that the order of the script is to first update Dynamodb and the elastic search, as ES syncs Dynamodb data it helps to keep things consistent if Dynamodb is updated first.

I switched the order.

File: aws/src/database/multi_table/amend_tenure.py (hunk 1 above)
Comment (adamtry) [reply aws/src/database/multi_table/amend_tenure.py]: Added a hint to the input just for clarity:
```python
UPDATE_DATE = parser.parse(timestr=input("Enter a date (e.g. DD/MM/YYYY or YYYY-MM-DD): ").strip(),
                           parserinfo=parser.parserinfo(dayfirst=True)).isoformat()
```

**Pull request merged**